doc/api18970.md

```diff
@@ -0,0 +1,175 @@
+# Draft material to review in support of [#18970][]
+
+[//]: # (cmark-gfm doc/api18970.md > /tmp/api18970.html)
+
+[#18970]: https://github.com/zephyrproject-rtos/zephyr/issues/18970
+
+## Background
+
+See also the Zephyr [glossary][].
+
+[glossary]: https://docs.zephyrproject.org/latest/glossary.html
+
+### Thread Terminology
```

Comment:
I'm guessing this is just background context that isn't going into the edits to the glossary in the final PR, right?

In other words, I expect the ultimate outcome of this work to be a set of new entries in the glossary, which cross-reference appropriately into the existing kernel docs which define these terms.

Reply:
This is simply the first step towards the complete requirements of #18970: not just the terminology, but the examples and policies that use that terminology.  Relates to your question about "queued" below: we'll need that term when we discuss the current/desired semantics of an `_async` suffix, perhaps to justify a `_queued` suffix for an API that is asynchronous and non-yielding.

Reply:
Marking this resolved for now, we can discuss this point further once it's a real docs PR.